### ISSUE TITLE
Failed to load baichuan with vllm

### ISSUE THREAD

**Issue by npuichigo** [bentoml/OpenLLM#309]
### Describe the bug

```
openllm start baichuan --model-id baichuan-inc/baichuan-13b-chat --backend vllm

2023-09-09T12:24:15+0800 [ERROR] [runner:llm-baichuan-runner:1] Traceback (most recent call last):
  File "/home/ichigo/miniconda3/envs/bento/lib/python3.10/site-packages/starlette/routing.py", line 677, in lifespan
    async with self.lifespan_context(app) as maybe_state:
  File "/home/ichigo/miniconda3/envs/bento/lib/python3.10/contextlib.py", line 199, in __aenter__
    return await anext(self.gen)
  File "/home/ichigo/miniconda3/envs/bento/lib/python3.10/site-packages/bentoml/_internal/server/base_app.py", line 75, in lifespan
    on_startup()
  File "/home/ichigo/miniconda3/envs/bento/lib/python3.10/site-packages/bentoml/_internal/runner/runner.py", line 317, in init_local
    raise e
  File "/home/ichigo/miniconda3/envs/bento/lib/python3.10/site-packages/bentoml/_internal/runner/runner.py", line 307, in init_local
    self._set_handle(LocalRunnerRef)
  File "/home/ichigo/miniconda3/envs/bento/lib/python3.10/site-packages/bentoml/_internal/runner/runner.py", line 150, in _set_handle
    runner_handle = handle_class(self, *args, **kwargs)
  File "/home/ichigo/miniconda3/envs/bento/lib/python3.10/site-packages/bentoml/_internal/runner/runner_handle/local.py", line 27, in __init__
    self._runnable = runner.runnable_class(**runner.runnable_init_params)  # type: ignore
  File "/home/ichigo/miniconda3/envs/bento/lib/python3.10/site-packages/openllm/_llm.py", line 1162, in __init__
    if not self.model: raise RuntimeError('Failed to load the model correctly (See traceback above)')
  File "/home/ichigo/miniconda3/envs/bento/lib/python3.10/site-packages/openllm/_llm.py", line 752, in model
    model = self.load_model(*self._model_decls, **self._model_attrs)
  File "/home/ichigo/miniconda3/envs/bento/lib/python3.10/site-packages/openllm/_assign.py", line 67, in inner
    raise OpenLLMException(f'Failed to initialise vLLMEngine due to the following error:\n{err}') from None
openllm_core.exceptions.OpenLLMException: Failed to initialise vLLMEngine due to the following error:
Failed to load the model config. If the model is a custom model not yet available in the HuggingFace transformers library, consider setting `trust_remote_code=True` in LLM or using the `--trust-remote-code` flag in the CLI.
```

#### Environment variable

```bash
BENTOML_DEBUG=''
BENTOML_QUIET=''
BENTOML_BUNDLE_LOCAL_BUILD=''
BENTOML_DO_NOT_TRACK=''
BENTOML_CONFIG=''
BENTOML_CONFIG_OPTIONS=''
BENTOML_PORT=''
BENTOML_HOST=''
BENTOML_API_WORKERS=''
```

#### System information

`bentoml`: 1.1.6
`python`: 3.10.12
`platform`: Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
`uid_gid`: 1000:1000
`conda`: 23.3.1
`in_conda_env`: True
<details><summary><code>conda_packages</code></summary>

<br>

```yaml
name: bento
channels:
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=5.1=1_gnu
  - bzip2=1.0.8=h7b6447c_0
  - ca-certificates=2023.08.22=h06a4308_0
  - ld_impl_linux-64=2.38=h1181459_1
  - libffi=3.4.4=h6a678d5_0
  - libgcc-ng=11.2.0=h1234567_1
  - libgomp=11.2.0=h1234567_1
  - libstdcxx-ng=11.2.0=h1234567_1
  - libuuid=1.41.5=h5eee18b_0
  - ncurses=6.4=h6a678d5_0
  - openssl=3.0.10=h7f8727e_2
  - pip=23.2.1=py310h06a4308_0
  - python=3.10.12=h955ad1f_0
  - readline=8.2=h5eee18b_0
  - setuptools=68.0.0=py310h06a4308_0
  - sqlite=3.41.2=h5eee18b_0
  - tk=8.6.12=h1ccaba5_0
  - tzdata=2023c=h04d1e81_0
  - wheel=0.38.4=py310h06a4308_0
  - xz=5.4.2=h5eee18b_0
  - zlib=1.2.13=h5eee18b_0
  - pip:
      - accelerate==0.22.0
      - anyio==3.7.1
      - appdirs==1.4.4
      - asgiref==3.7.2
      - attrs==23.1.0
      - backoff==2.2.1
      - bentoml==1.1.6
      - bitsandbytes==0.41.1
      - boto3==1.28.43
      - botocore==1.31.43
      - build==1.0.3
      - cattrs==23.1.2
      - circus==0.18.0
      - click==8.1.7
      - click-option-group==0.5.6
      - cloudpickle==2.2.1
      - cmake==3.27.4.1
      - coloredlogs==15.0.1
      - contextlib2==21.6.0
      - cpm-kernels==1.0.11
      - cuda-python==12.2.0
      - cython==3.0.2
      - deepmerge==1.1.0
      - deprecated==1.2.14
      - exceptiongroup==1.1.3
      - fastapi==0.103.1
      - fastcore==1.5.29
      - filetype==1.2.0
      - fs==2.4.16
      - fs-s3fs==1.1.1
      - ghapi==1.0.4
      - googleapis-common-protos==1.56.2
      - grpcio==1.58.0
      - grpcio-channelz==1.48.2
      - grpcio-health-checking==1.48.2
      - grpcio-reflection==1.48.2
      - h11==0.14.0
      - httpcore==0.17.3
      - httpx==0.24.1
      - huggingface-hub==0.16.4
      - humanfriendly==10.0
      - importlib-metadata==6.0.1
      - inflection==0.5.1
      - jinja2==3.1.2
      - jmespath==1.0.1
      - jsonschema==4.19.0
      - jsonschema-specifications==2023.7.1
      - lit==16.0.6
      - markdown-it-py==3.0.0
      - markupsafe==2.1.3
      - mdurl==0.1.2
      - mpmath==1.3.0
      - mypy-extensions==1.0.0
      - networkx==3.1
      - ninja==1.11.1
      - numba==0.57.1
      - numpy==1.24.4
      - nvidia-cublas-cu11==11.10.3.66
      - nvidia-cuda-cupti-cu11==11.7.101
      - nvidia-cuda-nvrtc-cu11==11.7.99
      - nvidia-cuda-runtime-cu11==11.7.99
      - nvidia-cudnn-cu11==8.5.0.96
      - nvidia-cufft-cu11==10.9.0.58
      - nvidia-curand-cu11==10.2.10.91
      - nvidia-cusolver-cu11==11.4.0.1
      - nvidia-cusparse-cu11==11.7.4.91
      - nvidia-nccl-cu11==2.14.3
      - nvidia-nvtx-cu11==11.7.91
      - openllm==0.3.3
      - openllm-client==0.3.3
      - openllm-core==0.3.3
      - opentelemetry-api==1.18.0
      - opentelemetry-exporter-jaeger==1.18.0
      - opentelemetry-exporter-jaeger-proto-grpc==1.18.0
      - opentelemetry-exporter-jaeger-thrift==1.18.0
      - opentelemetry-exporter-otlp==1.18.0
      - opentelemetry-exporter-otlp-proto-common==1.18.0
      - opentelemetry-exporter-otlp-proto-grpc==1.18.0
      - opentelemetry-exporter-otlp-proto-http==1.18.0
      - opentelemetry-exporter-zipkin==1.18.0
      - opentelemetry-exporter-zipkin-json==1.18.0
      - opentelemetry-exporter-zipkin-proto-http==1.18.0
      - opentelemetry-instrumentation==0.39b0
      - opentelemetry-instrumentation-aiohttp-client==0.39b0
      - opentelemetry-instrumentation-asgi==0.39b0
      - opentelemetry-instrumentation-grpc==0.39b0
      - opentelemetry-proto==1.18.0
      - opentelemetry-sdk==1.18.0
      - opentelemetry-semantic-conventions==0.39b0
      - opentelemetry-util-http==0.39b0
      - optimum==1.13.0
      - orjson==3.9.6
      - packaging==23.1
      - pathspec==0.11.2
      - pillow==10.0.0
      - pip-requirements-parser==32.0.1
      - pip-tools==7.3.0
      - prometheus-client==0.17.1
      - protobuf==3.20.3
      - psutil==5.9.5
      - pydantic==1.10.12
      - pynvml==11.5.0
      - pyparsing==3.1.1
      - pyproject-hooks==1.0.0
      - python-json-logger==2.0.7
      - python-multipart==0.0.6
      - pytz==2023.3.post1
      - pyyaml==6.0.1
      - pyzmq==25.1.1
      - ray==2.6.3
      - referencing==0.30.2
      - regex==2023.8.8
      - rich==13.5.2
      - rpds-py==0.10.2
      - s3transfer==0.6.2
      - safetensors==0.3.3
      - schema==0.7.5
      - scipy==1.11.2
      - sentencepiece==0.1.99
      - simple-di==0.1.5
      - six==1.16.0
      - sniffio==1.3.0
      - starlette==0.27.0
      - sympy==1.12
      - tabulate==0.9.0
      - thrift==0.16.0
      - tokenizers==0.13.3
      - tomli==2.0.1
      - torch==2.0.1
      - tornado==6.3.3
      - transformers==4.33.1
      - triton==2.0.0
      - urllib3==1.26.16
      - uvicorn==0.23.2
      - vllm==0.1.6
      - watchfiles==0.20.0
      - wrapt==1.15.0
      - xformers==0.0.21
      - zipp==3.16.2
prefix: /home/ichigo/miniconda3/envs/bento
```

</details>

<details><summary><code>pip_packages</code></summary>

<br>

```
accelerate==0.22.0
aiohttp==3.8.4
aiosignal==1.3.1
anyio==3.7.1
appdirs==1.4.4
asgiref==3.7.2
asttokens==2.2.1
async-timeout==4.0.2
attrs==23.1.0
audioread==3.0.0
backcall==0.2.0
backoff==2.2.1
bentoml==1.1.6
bitsandbytes==0.41.1
boto3==1.28.43
botocore==1.31.43
build==1.0.3
cattrs==23.1.2
certifi==2023.5.7
cffi==1.15.1
charset-normalizer==3.1.0
circus==0.18.0
click==8.1.7
click-option-group==0.5.6
cloudpickle==2.2.1
cmake==3.27.4.1
coloredlogs==15.0.1
contextlib2==21.6.0
cpm-kernels==1.0.11
cuda-python==12.2.0
Cython==3.0.2
datasets==2.12.0
decorator==5.1.1
deepmerge==1.1.0
Deprecated==1.2.14
dill==0.3.6
exceptiongroup==1.1.3
executing==1.2.0
fastapi==0.103.1
fastcore==1.5.29
filelock==3.12.0
filetype==1.2.0
frozenlist==1.3.3
fs==2.4.16
fs-s3fs==1.1.1
fsspec==2023.5.0
ghapi==1.0.4
googleapis-common-protos==1.56.2
grpcio==1.58.0
grpcio-channelz==1.48.2
grpcio-health-checking==1.48.2
grpcio-reflection==1.48.2
h11==0.14.0
httpcore==0.17.3
httpx==0.24.1
huggingface-hub==0.16.4
humanfriendly==10.0
idna==3.4
importlib-metadata==6.0.1
inflection==0.5.1
jedi==0.18.2
Jinja2==3.1.2
jmespath==1.0.1
joblib==1.2.0
jsonschema==4.19.0
jsonschema-specifications==2023.7.1
lazy_loader==0.2
librosa==0.10.0.post2
lit==16.0.6
llvmlite==0.40.0
markdown-it-py==3.0.0
MarkupSafe==2.1.3
matplotlib-inline==0.1.6
mdurl==0.1.2
mpmath==1.3.0
msgpack==1.0.5
multidict==6.0.4
multiprocess==0.70.14
mypy-extensions==1.0.0
networkx==3.1
ninja==1.11.1
numba==0.57.1
numpy==1.24.4
nvidia-cublas-cu11==11.10.3.66
nvidia-cuda-cupti-cu11==11.7.101
nvidia-cuda-nvrtc-cu11==11.7.99
nvidia-cuda-runtime-cu11==11.7.99
nvidia-cudnn-cu11==8.5.0.96
nvidia-cufft-cu11==10.9.0.58
nvidia-curand-cu11==10.2.10.91
nvidia-cusolver-cu11==11.4.0.1
nvidia-cusparse-cu11==11.7.4.91
nvidia-nccl-cu11==2.14.3
nvidia-nvtx-cu11==11.7.91
openllm==0.3.3
openllm-client==0.3.3
openllm-core==0.3.3
opentelemetry-api==1.18.0
opentelemetry-exporter-jaeger==1.18.0
opentelemetry-exporter-jaeger-proto-grpc==1.18.0
opentelemetry-exporter-jaeger-thrift==1.18.0
opentelemetry-exporter-otlp==1.18.0
opentelemetry-exporter-otlp-proto-common==1.18.0
opentelemetry-exporter-otlp-proto-grpc==1.18.0
opentelemetry-exporter-otlp-proto-http==1.18.0
opentelemetry-exporter-zipkin==1.18.0
opentelemetry-exporter-zipkin-json==1.18.0
opentelemetry-exporter-zipkin-proto-http==1.18.0
opentelemetry-instrumentation==0.39b0
opentelemetry-instrumentation-aiohttp-client==0.39b0
opentelemetry-instrumentation-asgi==0.39b0
opentelemetry-instrumentation-grpc==0.39b0
opentelemetry-proto==1.18.0
opentelemetry-sdk==1.18.0
opentelemetry-semantic-conventions==0.39b0
opentelemetry-util-http==0.39b0
optimum==1.13.0
orjson==3.9.6
packaging==23.1
pandas==2.0.1
parso==0.8.3
pathspec==0.11.2
pexpect==4.8.0
pickleshare==0.7.5
Pillow==10.0.0
pip-requirements-parser==32.0.1
pip-tools==7.3.0
pooch==1.6.0
prometheus-client==0.17.1
prompt-toolkit==3.0.38
protobuf==3.20.3
psutil==5.9.5
ptyprocess==0.7.0
pure-eval==0.2.2
pyarrow==12.0.0
pycparser==2.21
pydantic==1.10.12
Pygments==2.15.1
pynvml==11.5.0
pyparsing==3.1.1
pyproject_hooks==1.0.0
python-dateutil==2.8.2
python-json-logger==2.0.7
python-multipart==0.0.6
pytz==2023.3.post1
PyYAML==6.0.1
pyzmq==25.1.1
ray==2.6.3
referencing==0.30.2
regex==2023.8.8
requests==2.30.0
responses==0.18.0
rich==13.5.2
rpds-py==0.10.2
s3transfer==0.6.2
safetensors==0.3.3
schema==0.7.5
scikit-learn==1.2.2
scipy==1.11.2
sentencepiece==0.1.99
simple-di==0.1.5
six==1.16.0
sniffio==1.3.0
soundfile==0.12.1
soxr==0.3.5
stack-data==0.6.2
starlette==0.27.0
sympy==1.12
tabulate==0.9.0
threadpoolctl==3.1.0
thrift==0.16.0
tokenizers==0.13.3
tomli==2.0.1
torch==2.0.1
tornado==6.3.3
tqdm==4.65.0
traitlets==5.9.0
transformers==4.33.1
triton==2.0.0
typing_extensions==4.5.0
tzdata==2023.3
urllib3==1.26.16
uvicorn==0.23.2
vllm==0.1.6
watchfiles==0.20.0
wcwidth==0.2.6
wrapt==1.15.0
xformers==0.0.21
xmltodict==0.13.0
xxhash==3.2.0
yarl==1.9.2
zipp==3.16.2
```

</details>

**Comment by aarnphm** [comment]
Hi there, I'm investigating this. Thanks for reporting

**Comment by npuichigo** [comment]
@aarnphm Also cannot load baichuan 7B with pt backend.

I can directly load the model with transformers
```
from transformers import AutoModelForCausalLM, AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained("baichuan-inc/Baichuan-7B", trust_remote_code=True)
model = AutoModelForCausalLM.from_pretrained("baichuan-inc/Baichuan-7B", device_map="auto", trust_remote_code=True)
inputs = tokenizer('登鹳雀楼->王之涣\n夜雨寄北->', return_tensors='pt')
inputs = inputs.to('cuda:0')
pred = model.generate(**inputs, max_new_tokens=64,repetition_penalty=1.1)
print(tokenizer.decode(pred.cpu()[0], skip_special_tokens=True))
```

But these line will OOM 
```
openllm start baichuan --model-id baichuan-inc/baichuan-7b

openllm_core.exceptions.OpenLLMException: Failed to load tag=Tag(name='pt-baichuan-inc--baichuan-7b', version='c1a5c7d5b7f50ecc51bb0e08150a9f12e5656756') adapters_mapping=None config={'max_new_tokens': 2048, 'min_length': 0, 'early_stopping': False, 'num_beams': 1, 'num_beam_groups': 1, 'use_cache': True, 'temperature': 0.95, 'top_k': 50, 'top_p': 0.7, 'typical_p': 1.0, 'epsilon_cutoff': 0.0, 'eta_cutoff': 0.0, 'diversity_penalty': 0.0, 'repetition_penalty': 1.0, 'encoder_repetition_penalty': 1.0, 'length_penalty': 1.0, 'no_repeat_ngram_size': 0, 'renormalize_logits': False, 'remove_invalid_values': False, 'num_return_sequences': 1, 'output_attentions': False, 'output_hidden_states': False, 'output_scores': False, 'encoder_no_repeat_ngram_size': 0, 'n': 1, 'presence_penalty': 0.0, 'frequency_penalty': 0.0, 'use_beam_search': False, 'ignore_eos': False} runner_name='llm-baichuan-runner' model_id='baichuan-inc/baichuan-7b' backend='pt' into GPU: CUDA out of memory. Tried to allocate 64.00 MiB (GPU 0; 23.65 GiB total capacity; 23.14 GiB already allocated; 47.00 MiB free; 23.15 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
Tip: If you run into OOM issue, maybe try different offload strategy. See https://huggingface.co/docs/transformers/v4.31.0/en/main_classes/quantization#offload-between-cpu-and-gpu for more information.
```

**Comment by aarnphm** [comment]
Got it thanks for the information. I will take a look

**Comment by liulfy** [comment]
Is there any progress, please?

**Comment by aarnphm** [comment]
This has now been fixed and released on 0.4.28
